be/src/storage/rowset_update_state.cpp

```diff
@@ -103,6 +109,130 @@ Status RowsetUpdateState::_do_load(Rowset* rowset) {
     return Status::OK();
 }
 
+struct RowidSortEntry {
+    uint32_t rowid;
+    uint32_t idx;
+    RowidSortEntry(uint32_t rowid, uint32_t idx) : rowid(rowid), idx(idx) {}
+    bool operator<(const RowidSortEntry& rhs) const { return rowid < rhs.rowid; }
+};
+
+static void plan_read_by_rssid(const vector<uint64_t>& rowids, size_t* num_default,
+                               std::map<uint32_t, std::vector<uint32_t>>* rowids_by_rssid, vector<uint32_t>* pidxes) {
+    uint32_t n = rowids.size();
+    phmap::node_hash_map<uint32_t, vector<RowidSortEntry>> sort_entry_by_rssid;
+    std::vector<uint32_t> defaults;
+    for (uint32_t i = 0; i < n; i++) {
+        uint64_t v = rowids[i];
+        uint32_t rssid = v >> 32;
+        if (rssid == (uint32_t)-1) {
+            defaults.push_back(i);
+        } else {
+            uint32_t rowid = v & 0xffffffffUL;
```

Comment:
the mask is the ROWID_MASK in primary_index.cpp, if use the immediate number, it means write same code twice, reduce readability and maintainability, maybe can use a specialized function to handle these mask stuffs, make its' name more meaningful, to improve readability and maintainability.